interceptor.go

```diff
@@ -231,31 +140,73 @@ func (i *interceptor) WrapStreamingClient(next connect.StreamingClientFunc) conn
 			}
 		}
 		state := streamingState{
-			protocol: protocolToSemConv(conn.Peer()),
-			attrs:    attributesFromRequest(req),
+			protocol:   protocolToSemConv(conn.Peer().Protocol),
+			attributes: requestAttributes(req),
 		}
+		carrier := propagation.HeaderCarrier(conn.RequestHeader())
+		name := strings.TrimLeft(conn.Spec().Procedure, "/")
+		spanCtx := trace.SpanContextFromContext(ctx)
+		ctx = i.config.propagator.Extract(ctx, carrier)
+		ctx, span := i.config.tracer.Start(
+			trace.ContextWithRemoteSpanContext(ctx, spanCtx),
+			name,
+			trace.WithSpanKind(trace.SpanKindServer),
+			trace.WithAttributes(state.attributes...),
+		)
+		i.config.propagator.Inject(ctx, carrier)
 		return &streamingClientInterceptor{
 			StreamingClientConn: conn,
 			onClose: func() {
-				state.attrs = append(state.attrs, statusCodeAttribute(state.protocol, nil))
-				instr.duration.Record(ctx, i.config.now().Sub(requestStartTime).Milliseconds(), state.attrs...)
+				state.attributes = append(state.attributes, statusCodeAttribute(state.protocol, state.error))
+				span.SetAttributes(state.attributes...)
+				span.SetStatus(spanStatus(state.error))
+				span.End()
+				instrumentation.duration.Record(ctx, i.config.now().Sub(requestStartTime).Milliseconds(), state.attributes...)
 			},
 			receive: func(msg any, conn connect.StreamingClientConn) error {
-				return state.receive(ctx, instr, msg, conn)
+				size, err := state.receive(msg, conn)
+				if errors.Is(err, io.EOF) {
+					return err
+				}
+				span.AddEvent(messageKey,
+					trace.WithAttributes(
+						semconv.MessageTypeReceived,
+						semconv.MessageUncompressedSizeKey.Int(size),
+						semconv.MessageIDKey.Int(state.receivedCounter),
+					),
+				)
+				// In WrapStreamingClient the 'receive' is a response message.
+				instrumentation.responseSize.Record(ctx, int64(size), state.attributes...)
+				instrumentation.responsesPerRPC.Record(ctx, 1, state.attributes...)
```

Comment:
This is the part that had the bug before:
WrapStreamingClient:
- Receive -> Response
- Send -> Request

WrapStreamingHandler:
- Receive -> Request
- Send -> Response

And this means it can't be abstracted into `streamingState`